iOS/DuckDuckGo/AIChat/InputBox/SwitchBar/FlowManagers/KeyboardAdjustmentManager.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  KeyboardAdjustmentManager.swift
+//  DuckDuckGo
+//
+//  Copyright © 2025 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import UIKit
+
+/// Manages keyboard appearance and UI adjustments in response to keyboard notifications
+final class KeyboardAdjustmentManager {
+    
+    // MARK: - Types
+    
+    struct Configuration {
+        let logoOffset: CGFloat
+        let logoOffsetForVisibleKeyboard: CGFloat
+        let actionBarMargin: CGFloat
+        
+        static let `default` = Configuration(
+            logoOffset: 18,
+            logoOffsetForVisibleKeyboard: 50,
+            actionBarMargin: 16
+        )
+    }
+    
+    // MARK: - Properties
+    
+    private let configuration: Configuration
+    private let parentView: UIView
+    private weak var logoCenterYConstraint: NSLayoutConstraint?
+    private weak var actionBarBottomConstraint: NSLayoutConstraint?
+    
+    // MARK: - Initialization
+    
+    init(parentView: UIView,
+         logoCenterYConstraint: NSLayoutConstraint?,
+         actionBarBottomConstraint: NSLayoutConstraint?,
+         configuration: Configuration = .default) {
+        self.parentView = parentView
+        self.logoCenterYConstraint = logoCenterYConstraint
+        self.actionBarBottomConstraint = actionBarBottomConstraint
+        self.configuration = configuration
+        
+        setupKeyboardNotifications()
+    }
+    
+    deinit {
+        NotificationCenter.default.removeObserver(self)
+    }
+    
+    // MARK: - Private Methods
+    
+    private func setupKeyboardNotifications() {
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardWillShow(_:)),
+            name: UIResponder.keyboardWillShowNotification,
+            object: nil
+        )
+        
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardWillHide(_:)),
+            name: UIResponder.keyboardWillHideNotification,
+            object: nil
+        )
+    }
+    
+    @objc private func keyboardWillShow(_ notification: Notification) {
```

Comment:
I'm wondering why we're not using `UIKeyboardLayoutGuide` instead. It probably would be much simpler.

Reply:
No particular reason, this was extracted from other places in the app